Source/Core/Core/Core.cpp

```diff
@@ -855,6 +863,33 @@ void Callback_FramePresented(double actual_emulation_speed)
 // Called from VideoInterface::Update (CPU thread) at emulated field boundaries
 void Callback_NewField()
 {
+  if (!was_rng_seed_called)
+  {
+    LuaGameCubeButtonProbabilityEvent::SetRngSeeding();
+    was_rng_seed_called = true;
+  }
+
+  if (Lua::is_lua_script_active && !Lua::LuaEmu::waiting_for_save_state_load &&
+      !Lua::LuaEmu::waiting_for_save_state_save && !Lua::LuaEmu::waiting_to_start_playing_movie &&
+      !Lua::LuaEmu::waiting_to_save_movie)
+  {
+    for (int i = 0; i < 4; ++i)
+    {
+      Lua::LuaGameCubeController::overwrite_controller_at_specified_port[i] = false;
+      Lua::LuaGameCubeController::add_to_controller_at_specified_port[i] = false;
+      Lua::LuaGameCubeController::do_random_input_events_at_specified_port[i] = false;
+      Lua::LuaGameCubeController::random_button_events[i].clear();
+      Lua::LuaGameCubeController::button_lists_for_add_to_controller_inputs[i].clear();
+      memset(&Lua::LuaGameCubeController::new_overwrite_controller_inputs[i], 0,
+             sizeof(Movie::ControllerState));
+      memset(&Lua::LuaGameCubeController::add_to_controller_inputs[i], 0,
+             sizeof(Movie::ControllerState));
+    }
```

Comment:
This code should not live in core. 
Especially not a file named `Core/Core/Core.cpp`

First, Dolphin recently got an event hooking system. Take a look at `Common/HookableEvent.h` for the implementation, and `VideoCommon/VideoEvents.h` for both an example of how to use it, and some useful events:
* `BeforeFrameEvent`, just as the game starts rendering its the first object of the frame (gpu thread)
* `AfterFrameEvent`, Just the game marks the frame as fully rendered, but before the game is notified about this (gpu thread)
* `BeforePresentEvent`, Just before dolphin renders the frame to the screen. Perfect place to draw an overlay (currently gpu thread, but will move to a new present thread)
* `AfterPresentEvent`, should be just after the user can see the new presented frame on the screen (gpu/present thread)

You should be able to create a "Field-boundary" event type and trigger it here. 
Then register an event handler from your Lua subsystem.

Second, I mentione this before, but before random input stuff really should be done as a lua library rather than c++ code .